indexer.py

```diff
@@ -9,16 +9,21 @@
 from config import TDArgs
 from llama import Llama
 
-from test_dataset import collate_fn, UnittestDataset
+from test_dataset import (
+    collate_fn,
+    FileGranularityDataset,
+    FunctionGranularityDataset,
+)
 from torch.utils.data import DataLoader, DistributedSampler
 
 from transformers import AutoModelForCausalLM
 
 
 class Indexer:
-    def __init__(self, experiment_name):
+    def __init__(self, experiment_name, file: bool = False):
```

Comment:
Maybe let's have a enum for the support granularities. And name the argument accordingly (parsing_granularity or smth). Maybe we can leverage this enum when passing the arg via argparse as well so the. arg becomes `--granularity FILE`?

Reply:
Swapped parser to take in a granularity setting but used dictionary instead of enum so I can map to the correct class